client/src/components/Post/Post.tsx

```diff
@@ -140,6 +154,29 @@ class Post extends Component<Props, State> {
         this.props.history.push('/main/topic/' + this.state.topic.id, {topic : this.state.topic})
     }
 
+    handleEditClick = (index: number, content: string) => {
+        this.setState({toEditComment: index, editContent: content, editErrorMessage: ''})
+    }
+
+    handleEditCommentChange = (value: string) => {
+        this.setState({editContent: value})
+    }
+
+    handleCancelEdit = () => {
+        this.setState({editContent: '', toEditComment: -1})
+    }
+
+    handleEditSave = () => {
+        editComment(this.state.editContent, this.state.comments[this.state.toEditComment].id)
+            .then(res => {
+                const newComments = [...this.state.comments]
+                newComments[this.state.toEditComment] = res
+                this.setState({comments: newComments, editErrorMessage: ''})
+                this.handleCancelEdit()
+            })
+            .catch(() => this.setState({editErrorMessage: 'An error occurred editing your post'}))
```

Comment:
@Nestiii fix error message
```suggestion
            .catch(() => this.setState({editErrorMessage: 'An error occurred editing your comment'}))
```